lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -411,6 +411,86 @@ LinearLayout hopperMmaToLinearLayout(ArrayRef<int64_t> shape,
   return combineCtaCgaWithShape(ctaLayout, mma.getCTALayout(), shape);
 }
 
+LinearLayout mfmaToLinearLayout(ArrayRef<int64_t> shape,
+                                AMDMfmaEncodingAttr mfma) {
+  int rank = shape.size();
+  assert(rank == mfma.getWarpsPerCTA().size());
+
+  bool hasBatchDim = rank == 3;
+  int mIndex = 0 + hasBatchDim;
+  int nIndex = 1 + hasBatchDim;
+  (void)mIndex, (void)nIndex;
+
+  assert(((shape[mIndex] == 1 || shape[mIndex] >= mfma.getMDim()) &&
+          (shape[nIndex] == 1 || shape[nIndex] >= mfma.getNDim())) &&
+         "Unsupported tensor shape for given mfma layout");
+
+  assert(((mfma.getMDim() == 32 && mfma.getNDim() == 32) ||
+          (mfma.getMDim() == 16 && mfma.getNDim() == 16)) &&
+         "Unsupported mfma type");
+
+  MLIRContext *ctx = mfma.getContext();
+  SmallVector<StringAttr> outDimNames = standardOutDimNames(ctx, rank);
+
+  StringAttr rDim = S("register");
+  StringAttr lDim = S("lane");
+  StringAttr wDim = S("warp");
```

Comment:
Nit, elsewhere I was using the names "kRegister", "kDim", and "kWarp" for these variables.

Reply:
done